### ISSUE TITLE
loss computation

### ISSUE THREAD

**Issue by hachikong** [mahmoodlab/PORPOISE#3]
Hi, thanks a lot for your sharing your code.

I have a question on the following code:

https://github.com/mahmoodlab/PORPOISE/blob/8ddca09b6812df7a0a643126a8b08cc3ec2e7d23/utils/loss_func.py#L104

    neg_l = censored_loss + uncensored_loss
    if alpha is not None:
        loss = (1 - alpha) * neg_l + alpha * uncensored_loss

which can be split into:

    loss = (1 - alpha) * (censored_loss + uncensored_loss) + alpha * uncensored_loss
         = (1 - alpha) * censored_loss + uncensored_loss - alpha * uncensored_loss + alpha * uncensored_loss
         = (1 - alpha) * censored_loss + uncensored_loss

Is this reasonable?

**Comment by Richarizardd** [comment]
What do you mean by reasonable? I think you need to distribute `(1-alpha)` over the other terms in Line 2.

**Comment by hachikong** [comment]
Hi @Richarizardd , thanks a lot for your kind reply!

```
loss 
= (1 - alpha) * neg_l + alpha * uncensored_loss
= (1 - alpha) * (censored_loss + uncensored_loss) + alpha * uncensored_loss
= (1 - alpha) * censored_loss + (1 - alpha) * uncensored_loss + alpha * uncensored_loss
= (1 - alpha) * censored_loss + (1 - alpha + alpha) * uncensored_loss
= (1 - alpha) * censored_loss + uncensored_loss
```
So, it seems the alpha is not applied to the `uncensored_loss`. I'm not sure if this is reasonable, as in the paper, the `alpha` should be applied to `uncensored_loss`.

**Comment by Richarizardd** [comment]
Hi @hachikong - Our implementation follows that of [Zadeh and Schmid](https://pubmed.ncbi.nlm.nih.gov/32149626/), which we believe is reasonable. You can see the negative log-likelihood term described in Equation 6, with the 2nd loss term (Equation 14) described at the contribution of uncensored instances in the negative likelihood.

**Comment by hachikong** [comment]
Hi @Richarizardd , thanks a lot for your kind clarification! Yeah, it seems your implementation is different from theirs'. In your implementation, the network output is the hazard function, while their network output is the survival function. Both losses are the negative log-likelihood. I will read the paper more carefully. Thank you!
